core/src/main/scala/org/bitcoins/core/script/crypto/CryptoInterpreter.scala

```diff
@@ -451,43 +451,46 @@ sealed abstract class CryptoInterpreter {
           val flags = updatedProgram.flags
           val requireMinimal =
             ScriptFlagUtil.requireMinimalData(updatedProgram.flags)
+
           val numT = ScriptNumber(updatedProgram.stack(1).bytes, requireMinimal)
 
-          if (numT.isFailure) {
-            throw numT.failed.get
-          }
-          val restOfStack =
-            updatedProgram.stack.tail.tail.tail //remove signature, num, pubkey
-
-          val tapscriptE = evalChecksigTapscript(updatedProgram)
-          tapscriptE match {
-            case Left(err) =>
-              if (
-                err == ScriptErrorDiscourageUpgradablePubkeyType && !ScriptFlagUtil
-                  .discourageUpgradablePublicKey(flags)
-              ) {
-                //trivially pass signature validation as required by BIP342
-                //when the public key type is not known and the
-                //SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_PUBKEYTYPE is NOT set
+          if (numT.isFailure || numT.get.byteSize > 4) {
+            // not sure what error to throw here
+            updatedProgram.failExecution(ScriptErrorInvalidStackOperation)
```

Comment:
Not sure what error to throw here, this seemed the most logical

Reply:
Non minimally encoded numbers get `ScriptErrorUnknownError` since an actual exception throw in the interpreter. I don't make the rules 🤷‍♂️ 

https://github.com/bitcoin-s/bitcoin-s/blob/0397ca6c78ed307c25be90edf4306ecfc47824f1/core/src/main/scala/org/bitcoins/core/script/crypto/CryptoInterpreter.scala#L336